labcodes/labstorybook
# DSYS6/buttons-js

**Link to task:**
https://labcodes.atlassian.net/browse/DSYS-6

**How to test:**
- With the _LabSystem_ running, go to the `Component/Button` div in the _LabStoryBook_, then check if all components are rendered as expected.
- Locally, from `labstorybook` dir, run `npm run test -- -u`, then check if all tests are passing.

**Description of your solution:**
This PR refactor the architecture of `Button` and introduces `AbstractButton`. The later should serve as a "private" component, to be used for building "public" `[Default]Button`, `TextButton` and `OutlineButton` components through the `variant` props. Support for passing click-handlers as props to `Button`-related components are also enabled. `Button.stories` is refactored accordingly, with minor syntax fixes. Some snapshot-based tests are implemented as well.
